c-dependencies/js-compute-runtime/js-compute-builtins.cpp

```diff
@@ -3238,6 +3282,583 @@ namespace FetchEvent {
+
+    RootedValue key_val(cx);
+    RootedValue val_val(cx);
+
+    if (type != ITERTYPE_VALUES) {
+      auto chars = JS::UTF8Chars((char*)param.name.data, param.name.len);
+      RootedString str(cx, JS_NewStringCopyUTF8N(cx, chars));
+      if (!str) return false;
+      key_val = JS::StringValue(str);
+    }
+
+    if (type != ITERTYPE_KEYS) {
+      auto chars = JS::UTF8Chars((char*)param.value.data, param.value.len);
+      RootedString str(cx, JS_NewStringCopyUTF8N(cx, chars));
+      if (!str) return false;
+      val_val = JS::StringValue(str);
+    }
+
+    RootedValue result_val(cx);
+
+    switch (type) {
+      case ITERTYPE_ENTRIES: {
+        RootedObject pair(cx, JS::NewArrayObject(cx, 2));
+        if (!pair) return false;
+        JS_DefineElement(cx, pair, 0, key_val, JSPROP_ENUMERATE);
+        JS_DefineElement(cx, pair, 1, val_val, JSPROP_ENUMERATE);
+        result_val = JS::ObjectValue(*pair);
+        break;
+      }
+      case ITERTYPE_KEYS: {
+        result_val = key_val;
+        break;
+      }
+      case ITERTYPE_VALUES: {
+        result_val = val_val;
+        break;
+      }
+      default:
+        MOZ_RELEASE_ASSERT(false, "Invalid iter type");
+    }
+
+    JS_DefineProperty(cx, result, "value", result_val, JSPROP_ENUMERATE);
+
+    JS::SetReservedSlot(self, Slots::Index, JS::Int32Value(index + 1));
+    args.rval().setObject(*result);
+    return true;
+  }
+
+  const JSFunctionSpec methods[] = {
+    JS_FN("next", next, 0, JSPROP_ENUMERATE),
+  JS_FS_END};
+
+  const JSPropertySpec properties[] = {JS_PS_END};
+
+  CLASS_BOILERPLATE_CUSTOM_INIT(URLSearchParamsIterator)
+
+  bool init_class(JSContext* cx, HandleObject global) {
+    RootedObject iterator_proto(cx, JS::GetRealmIteratorPrototype(cx));
+    if (!iterator_proto) return false;
+
+    if (!init_class_impl(cx, global, iterator_proto))
+      return false;
+
+    // Delete both the `URLSearchParamsIterator` global property and the `constructor`
+    // property on `URLSearchParamsIterator.prototype`.
+    // The latter because Iterators don't have their own constructor on the prototype.
+    return JS_DeleteProperty(cx, global, class_.name) &&
+           JS_DeleteProperty(cx, proto_obj, "constructor");
+  }
+
+  JSObject* create(JSContext* cx, HandleObject params, uint8_t type) {
+    MOZ_RELEASE_ASSERT(type <= ITERTYPE_VALUES);
+
+    RootedObject self(cx, JS_NewObjectWithGivenProto(cx, &class_, proto_obj));
+    if (!self) return nullptr;
+
+    JS::SetReservedSlot(self, Slots::Params, JS::ObjectValue(*params));
+    JS::SetReservedSlot(self, Slots::Type, JS::Int32Value(type));
+    JS::SetReservedSlot(self, Slots::Index, JS::Int32Value(0));
+
+    return self;
+  }
+}
+
+
+namespace URLSearchParams {
+  namespace Slots { enum {
+    Url,
+    Params,
+    Count
+  };};
+
+  namespace detail {
+    bool append(JSContext* cx, HandleObject self, HandleValue key, HandleValue val, const char* fun_name) {
```

Comment:
`fun_name` is unused here -- was it meant to go into an error message of some sort maybe?

Reply:
ah, I forgot to address this. The parameter is required because `append` is used as a template parameter in `init_class`. I renamed it to `_` to make clear it's unused, though.